hpx/runtime/threads/detail/scheduling_loop.hpp

```diff
@@ -672,6 +672,7 @@ namespace hpx { namespace threads { namespace detail
                 {
                     // clean up terminated threads one more time before sleeping
                     bool can_exit =
+                        !running &&
                         scheduler.SchedulingPolicy::cleanup_terminated(
```

Comment:
Is it intentional to call `cleanup_terminated` only if `!running` evaluates to `true`?

Reply:
Not 100%. It could be either way, but I put it first since when `running` is `true` the "idle" `cleanup_terminated` will be called eventually if there's no work to do (at the end of `scheduling_loop`).